src/main/java/net/schmizz/sshj/sftp/RemoteDirectory.java

```diff
@@ -25,7 +25,7 @@
 public class RemoteDirectory
         extends RemoteResource {
 
-    public RemoteDirectory(Requester requester, String path, byte[] handle) {
+    public RemoteDirectory(SFTPEngine requester, String path, byte[] handle) {
```

Comment:
I added a protected getLoggingFactory to SFTPEngine.  If you’d prefer I could add the method to the Requester interface.  Since SFTPEngine, RemoteDirectory and RemoteFile are all in the same package, it seemed to me that treating the Requester like an SFTPEngine in the constructors would be harmless.

> On Aug 24, 2016, at 6:40 AM, Jeroen van Erp <notifications@github.com> wrote:
> 
> In src/main/java/net/schmizz/sshj/sftp/RemoteDirectory.java <https://github.com/hierynomus/sshj/pull/267#discussion_r76040773>:
> 
> > @@ -25,7 +25,7 @@
> >  public class RemoteDirectory
> >          extends RemoteResource {
> >  
> > -    public RemoteDirectory(Requester requester, String path, byte[] handle) {
> > +    public RemoteDirectory(SFTPEngine requester, String path, byte[] handle) {
> Why?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/hierynomus/sshj/pull/267/files/f63a88ec9f1ca21db2d18eb68056fcf51a107ff5#r76040773>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABIPfdMQUVOMsx1BZaT3pbe7nskVQFDnks5qjC2mgaJpZM4JrhJ->.
>